crates/revm/src/instructions/macros.rs

```diff
@@ -53,6 +53,10 @@ macro_rules! memory_resize {
         if let Some(new_size) =
             crate::interpreter::memory::next_multiple_of_32(offset.saturating_add(len))
         {
+            if new_size > crate::interpreter::memory::MEMORY_LIMIT {
+                return Return::FatalNotSupported;
```

Comment:
The behaviour recommended in EIP-1985 is to go out of gas.

Note: solidity has runtime code for reverting if your memory allocation extends the free memory pointer more than `2**64 -1`.